tests/v2/validation/auth/userretention/userretention.go

```diff
@@ -0,0 +1,166 @@
+	log.Infof("Getting role bindings for user %s in cluster %s", userID, clusterID)
+	listOpt := v1.ListOptions{}
+	roleBindings, err := rbac.ListRoleBindings(rancherClient, clusterID, "", listOpt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch RoleBindings: %w", err)
+	}
+
+	var userRoleBindings []rbacv1.RoleBinding
+	for _, rb := range roleBindings.Items {
+		for _, subject := range rb.Subjects {
+			if subject.Name == userID {
+				userRoleBindings = append(userRoleBindings, rb)
+				break
+			}
+		}
+	}
+	log.Infof("Found %d role bindings for user %s", len(userRoleBindings), userID)
+	return userRoleBindings, nil
+}
+
+func getBindings(rancherClient *rancher.Client, userID string) (map[string]interface{}, error) {
+	log.Infof("Getting all bindings for user %s", userID)
+	bindings := make(map[string]interface{})
+
+	roleBindings, err := getRoleBindings(rancherClient, rbac.LocalCluster, userID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get role bindings: %w", err)
+	}
+	bindings["RoleBindings"] = roleBindings
+
+	log.Info("Getting cluster role bindings")
+	clusterRoleBindings, err := rbac.ListClusterRoleBindings(rancherClient, rbac.LocalCluster, v1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list cluster role bindings: %w", err)
+	}
+	bindings["ClusterRoleBindings"] = clusterRoleBindings.Items
+
+	log.Info("Getting global role bindings")
+	globalRoleBindings, err := rancherClient.Management.GlobalRoleBinding.ListAll(&types.ListOpts{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list global role bindings: %w", err)
+	}
+	bindings["GlobalRoleBindings"] = globalRoleBindings.Data
+
+	log.Info("Getting cluster role template bindings")
+	clusterRoleTemplateBindings, err := rancherClient.Management.ClusterRoleTemplateBinding.List(&types.ListOpts{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list cluster role template bindings: %w", err)
+	}
+	bindings["ClusterRoleTemplateBindings"] = clusterRoleTemplateBindings.Data
+
+	log.Info("All bindings retrieved successfully")
+	return bindings, nil
+}
+
+func pollUserStatus(rancherClient *rancher.Client, userID string, expectedStatus bool) error {
+	log.Infof("Polling user status for user %s, expected status: %v", userID, expectedStatus)
+	ctx, cancel := context.WithTimeout(context.Background(), defaultWaitDuration)
+	defer cancel()
+
+	return wait.PollUntilContextTimeout(ctx, pollInterval, defaultWaitDuration, true, func(ctx context.Context) (bool, error) {
+		log.Info("Logging in with default admin user")
+		adminID, err := users.GetUserIDByName(rancherClient, "admin")
+		if err != nil {
+			return false, fmt.Errorf("failed to get admin user ID: %v", err)
+		}
+		adminUser, err := rancherClient.Management.User.ByID(adminID)
+		if err != nil {
+			return false, fmt.Errorf("failed to get admin user: %v", err)
+		}
+		adminUser.Password = rancherClient.RancherConfig.AdminPassword
```

Comment:
can you move this outside the polling function as the admin user information doesn’t need to be fetched repeatedly?